drivers/gles3/shaders/canvas.glsl

```diff
@@ -583,16 +583,15 @@ void main() {
 
 #if !defined(COLOR_USED)
 	//default behavior, texture by color
-
-#ifdef USE_DISTANCE_FIELD
-	const float smoothing = 1.0 / 32.0;
-	float distance = textureLod(color_texture, uv, 0.0).a;
-	color.a = smoothstep(0.5 - smoothing, 0.5 + smoothing, distance) * color.a;
-#else
 	color *= texture(color_texture, uv);
-
 #endif
 
+#ifdef USE_DISTANCE_FIELD
+	// Higher is smoother, but also more blurry. Lower is crisper, but also more aliased.
+	// TODO: Adjust automatically based on screen resolution/font size ratio.
+	const float smoothing = 0.125;
+	float dist = texture(color_texture, uv, 0.0).a;
+	color.a = smoothstep(0.5 - smoothing, 0.5 + smoothing, dist);
 #endif
```

Comment:
I'm a bit skeptical of the shader change here.  This logic results in 2 changes of behaviour that seem unintended:
1. If `COLOR` is used, then the font texture is still sampled which changes behavior (although seems like an improvement to me)
2. The RGB color from the distance field texture is used now even when using distance field rendering. It seems like this works because the texture is always white, but its probably best to just not have the texture sample at all


In 4.x the MSDF rendering is also an if/else combined with the texture sample.

Reply:
@Calinou as this is his code. I'll try and take a look too.

Reply:
I must admit I'm slightly lost and not so familiar with this part of the shader, and defer to you and Calinou (and the situation where COLOR is in use etc) but would this be any better? (And presumably the same for GLES2)

I was interpreting your post above as "when `color_texture` is being used for SDF, only the alpha is used, and the RGB is all white, so no need to read them." Whether this makes a difference in practice I have no idea :grin:  (as the read is probably always RGBA).

```
#ifdef USE_DISTANCE_FIELD
	// Higher is smoother, but also more blurry. Lower is crisper, but also more aliased.
	// TODO: Adjust automatically based on screen resolution/font size ratio.
	const float smoothing = 0.125;
	float dist = texture(color_texture, uv, 0.0).a;
	color.a = smoothstep(0.5 - smoothing, 0.5 + smoothing, dist);
#else
#if !defined(COLOR_USED)
	//default behavior, texture by color
	color *= texture(color_texture, uv);
#endif
#endif
```
Here the SDF would be applied to color even if `COLOR_USED`, and the RGB is not applied when using distance field.

Reply:
That makes more sense to me and is much closer to what we do in 4.x